mapie/tests/test_regression.py

```diff
@@ -867,14 +881,104 @@ def early_stopping_monitor(i, est, locals):
         else:
             return False
 
-    mapie.fit(X, y, monitor=early_stopping_monitor)
+    mapie.fit(X, y, fit_params={'monitor': early_stopping_monitor})
 
     assert mapie.estimator_.single_estimator_.estimators_.shape[0] == 3
 
     for estimator in mapie.estimator_.estimators_:
         assert estimator.estimators_.shape[0] == 3
 
 
+def test_predict_parameters_passing() -> None:
+    """
+    Test passing predict parameters.
+    Checks that y_pred from train are 0 and y_pred from test are 0
+    """
+
+    custom_gbr = CustomGradientBoostingRegressor(random_state=random_state)
+
+    X_train, X_test, y_train, y_test = (
+        train_test_split(X, y, test_size=0.2, random_state=random_state)
+    )
+    custom_gbr = CustomGradientBoostingRegressor(random_state=random_state)
+    mapie_1 = MapieRegressor(estimator=custom_gbr)
+    mapie_2 = MapieRegressor(estimator=custom_gbr)
+    predict_params = {'check_predict_params': True}
+    mapie_1 = mapie_1.fit(
+        X_train, y_train, predict_params=predict_params
+    )
+    mapie_2 = mapie_2.fit(X_train, y_train)
+    y_pred_1 = mapie_1.predict(X_test, **predict_params)
+    y_pred_2 = mapie_2.predict(X_test)
+    np.testing.assert_allclose(y_pred_1, 0)
+    np.testing.assert_allclose(mapie_1.conformity_scores_, np.abs(y_train))
+    with np.testing.assert_raises(AssertionError):
+        np.testing.assert_array_equal(y_pred_1, y_pred_2)
+
+
+def test_fit_and_predict_parameters_passing() -> None:
```

Comment:
There are a lot of tests all at once, please split the tests into multiple tests. Do one test to check for the size of the estimators and one to check that the results are the same.

Reply:
Done